db/seeds.rb

```diff
@@ -1,7 +1,104 @@
-# This file should contain all the record creation needed to seed the database with its default values.
-# The data can then be loaded with the rails db:seed command (or created alongside the database with db:setup).
 #
-# Examples:
+# This is data randomly generated using Faker gem to support testing in Alpha. Some of the data is
+# fixed to aid testing and portion will be randomly re-generated every time the seed is re-run.
 #
-#   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
-#   Character.create(name: 'Luke', movie: movies.first)
+foster_parents = [
+  {
+    id: 101,
+    email: "r.johnston@example.com",
+    first_name: "Rodger",
+    last_name: "Johnston"
+  },
+  {
+    id: 102,
+    email: "lucy82@example.com",
+    first_name: "Lucy",
+    last_name: "Casper"
+  }
+]
+(103..110).each do |id|
+  foster_parents << {
+    id: id,
+    email: Faker::Internet.safe_email,
+    first_name: Faker::Name.first_name,
+    last_name: Faker::Name.last_name
+  }
+end
+
+children = [
+  {
+    id: 201,
+    first_name: "Fredrick",
+    last_name: "Gutmann"
+  },
+  {
+    id: 202,
+    first_name: "Yasmin",
+    last_name: "Kub"
+  }
+]
+(203..210).each do |id|
+  children << {
+    id: id,
+    first_name: Faker::Name.first_name,
+    last_name: Faker::Name.last_name
+  }
+end
+
+placements = [
+  {
+    id: 301,
+    foster_parent_id: 102,
+    child_id: 202
+  }
+]
+
+diary_logs = 10.times.map do |i|
+  {
+    placement_id: 301,
+    note: Faker::Lorem.paragraph,
+    created_at: i.days.ago,
+    updated_at: i.days.ago
+  }
+end
+
+# data insertion for the above data
+
+foster_parents.each do |fp|
+  user = User.create!(
```

Comment:
could you use create_or_find_by ?

Reply:
We could do that, but I think this seed is really going to be run on a truncated database only. Particularly if we are going to be using Faker on some parts of it.

Reply:
Btw. feel free to change and adapt this as we go, I think that's the main advantage of having this checked in with the rest of the code. Each PR can make changes here which are adequate for testing..